exporter/collector/logs.go

```diff
@@ -349,7 +382,14 @@ func (l logMapper) logToSplitEntries(
 	if log.SeverityNumber() < 0 || int(log.SeverityNumber()) > len(severityMapping)-1 {
 		return []logging.Entry{entry}, fmt.Errorf("Unknown SeverityNumber %v", log.SeverityNumber())
 	}
-	entry.Severity = severityMapping[log.SeverityNumber()]
+	logSeverity := severityMapping[log.SeverityNumber()]
```

Comment:
The nested boolean logic is a bit hard for me to follow. Would you consider instead:

```go
func cloudLoggingSeverity(log plog.LogRecord) logging.Severity {
  severityNumber := log.SeverityNumber()
  if severityNumber == 0 {
    severityNumber := otelSeverityTexts[strings.ToLower(log.SeverityText())]
  }
  return severityMapping[severityNumber]
}
```

The double negative in the comment is also difficult, could you rephrase it? e.g., document how severity mapping works overall, rather than this particular sub-case.